sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -225,6 +229,13 @@ ssize_t sock_udp_recv_buf(sock_udp_t *sock, void **data, void **buf_ctx,
         memcpy(remote, &tmp, sizeof(tmp));
         remote->port = byteorder_ntohs(hdr->src_port);
     }
+    if (local != NULL) {
+        gnrc_pktsnip_t *ipv6 = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_IPV6);
```

Comment:
Might be worth to extend `gnrc_sock_recv()` to also provide the destination. Otherwise, we iterate `pkt` unnecessary twice.